server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -520,11 +505,9 @@ public SnapshotsInProgress(StreamInput in) throws IOException {
             if (in.getVersion().onOrAfter(METADATA_FIELD_INTRODUCED)) {
                 userMetadata = in.readMap();
             }
-            final boolean useShardGenerations;
-            if (in.getVersion().onOrAfter(SnapshotsService.SHARD_GEN_IN_REPO_DATA_VERSION)) {
-                useShardGenerations = in.readBoolean();
-            } else {
-                useShardGenerations = false;
+            final Version version = in.getVersion();
+            if (version.before(Version.V_8_0_0) && version.onOrAfter(SnapshotsService.SHARD_GEN_IN_REPO_DATA_VERSION)) {
```

Comment:
given that this logic can only be removed in 9.0, perhaps simpler to just unconditionally do a  `in.readBoolean();` here and add an assertion that fails once we branch 9.0. Also note that `version.onOrAfter(SnapshotsService.SHARD_GEN_IN_REPO_DATA_VERSION)` should be unnecessary once we stop testing master against 7.5. This is also why I prefer to wait with PRs like this until that has happened, it simplifies the BWC conditions.

Reply:
>  This is also why I prefer to wait with PRs like this until that has happened, it simplifies the BWC conditions.

Right that's a fair point. I'll just close here then and look into this again once `7.5` is out :)